src/electron/adapters/electron-storage-adapter.ts

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import { IndexedDBDataKeys } from '../../background/IndexedDBDataKeys';
+import { StorageAdapter } from '../../common/browser-adapters/storage-adapter';
+import { IndexedDBAPI } from '../../common/indexedDB/indexedDB';
+import { createDefaultLogger } from '../../common/logging/default-logger';
+import { Logger } from '../../common/logging/logger';
+
+export class ElectronStorageAdapter implements StorageAdapter {
+    constructor(private readonly indexedDBInstance: IndexedDBAPI, private logger: Logger = createDefaultLogger()) {}
+
+    public setUserData(items: Object, callback?: () => void): void {
+        this.indexedDBInstance
+            .setItem(IndexedDBDataKeys.installation, items)
+            .then(() => callback)
+            .catch(error => {
+                this.logger.error('Error occured when trying to set user data:', error);
+            });
+    }
+
+    public getUserData(keys: string | Object | string[], callback: (items: { [key: string]: any }) => void): void {
+        this.indexedDBInstance
+            .getItem(IndexedDBDataKeys.installation)
+            .then(data => {
+                return callback(data);
```

Comment:
should this `then` call:
```
 .then(data => {
                return callback(data);
```
look the same as the previous one?
``` .then(() => callback)```

Reply:
Updated.